index.ts

```diff
@@ -361,18 +366,24 @@ export class CapabilitiesController extends BaseController<any, any> implements
   /**
    * Used for granting a new set of permissions,
    * after the user has approved it.
-   * 
+   *
    * @param {string} domain - The domain receiving new permissions.
    * @param {IRequestedPermissions} approvedPermissions - An object of objects describing the granted permissions.
    * @param {JsonRpcResponse} res - The response.
    * @param {JsonRpcEngineEndCallback} end - The end function.
    */
   grantNewPermissions (
     domain: string,
-    approved: IRequestedPermissions, 
+    approved: IRequestedPermissions,
     res: JsonRpcResponse<any>,
     end: JsonRpcEngineEndCallback
   ): void {
+
+    if (!domain || typeof domain !== 'string') {
+      res.error = invalidReq(`Invalid domain: '${domain}'.`);
+      return end(res.error);
+    }
+
```

Comment:
Added this check because `grantNewPermissions` is called in one place in the extension, for exposing accounts to legacy dapps.